### PR TITLE
Clean up secrets created during the tests.

### PR DESCRIPTION
**Pull request by dargudear-google**
I am working on-recreating a prow job and these prow jobs deleted during migration to community infra. [Discussion ref](https://groups.google.com/a/kubernetes.io/g/dev/c/p6PAML90ZOU).

I am re-enabling this [job](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/secrets-store-csi-driver/secrets-store-csi-driver-config.yaml#L682-L725) to run on community infra. 
To run the job in community infra, I need to create secrets from google secret manager.

This PR is to remove secrets created during the job execution.

### CODE REVIEW

File: cmd/janitor/gcp_janitor.py [hunk 1]
```diff
@@ -588,6 +588,51 @@ def clean_gke_cluster(project, age, filt):
...
+        if 'name' not in item or 'createTime' not in item:
+            raise ValueError('name and createTime must be present: %r' % item)
+
+        # createTime in format '2024-10-16T07:22:22.119486Z'
+        # Only `024-10-16T07:22:22` this part is required.
+        item['createTime'] = item['createTime'].split('+')[0]
+        created = datetime.datetime.strptime(
+            item['createTime'][:19], '%Y-%m-%dT%H:%M:%S')
+        if created < age:
+            log('Found stale gke cluster %r, created time = %r' %
```
Comment (krzyzacy) [inline cmd/janitor/gcp_janitor.py:620]: s/cluster/secret

**Comment by dargudear-google** [comment]
> does this work locally? could you update the description for your use case?

Yes. I tried with my personal project and secret cleaning was successful with one related error

```
Fail to list resource 'topics' from project '<project_id>': ValueError("missing key: creationTimestamp or createTime
```

**Review by krzyzacy — APPROVED** [review]

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/k8s.io/pull/7531 broke Kubernetes CI which reports containing https://github.com/kubernetes-sigs/boskos/compare/bc478df...445d89e

that appears to solely contain this PR

I think we need to fix and/or revert.

tentatively: it fails because it tries to use the secret manager API even if it's not enabled, leaving the projects stuck in dirty state and causing us to run out of projects.

we need to gracefully handle that the project doesn't have the API enabled and therefore has nothing to clean up.

**Comment by BenTheElder** [comment]
filed https://github.com/kubernetes-sigs/boskos/issues/209 to track next steps.

**Comment by krzyzacy** [comment]
we probably should revert the deployment first

**Comment by BenTheElder** [comment]
> we probably should revert the deployment first

That's done and CI is green again, I did that first before following up here (https://github.com/kubernetes/test-infra/issues/33910), but we will want to get boskos into a state where updating the deployment is safe.

**Comment by dargudear-google** [comment]
Here is the PR with API enabled check https://github.com/kubernetes-sigs/boskos/pull/211

**Pull request merged**